tasks/rh-sign-cosign-image/rh-sign-cosign-image.yaml

```diff
@@ -0,0 +1,103 @@
+      type: string
+  workspaces:
+    - name: data
+      description: workspace to read and save files
+  steps:
+    - name: sign-image
+      image: quay.io/konflux-ci/release-service-utils:38c3bfd7479c86b832cba5b61f9bbde40c469393
+      env:
+        - name: AWS_REGION
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_REGION
+        - name: AWS_ACCESS_KEY_ID
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_ACCESS_KEY_ID
+        - name: AWS_SECRET_ACCESS_KEY
+          valueFrom:
+            secretKeyRef:
+              name: $(params.secretName)
+              key: AWS_SECRET_ACCESS_KEY
+      script: |
+        #!/usr/bin/env bash
+        set -eu
+        function fetch_cosign_key() {
+          echo $(kubectl_get_arg_from_config_map $1 SIG_KEY)
+        } 
+
+        SNAPSHOT_PATH=$(workspaces.data.path)/$(params.snapshotPath)
+
+        DATA_FILE="$(workspaces.data.path)/$(params.dataPath)"
+        if [ ! -f "${DATA_FILE}" ] ; then
+            echo "No valid data file was provided."
+            exit 1
+        fi
+
+        request=$(jq -r '.sign.request // "hacbs-signing-pipeline"' ${DATA_FILE})
+        CONFIG_MAP_NAME=$(jq -rc '.sign.configMapName // "signing-config-map"' ${DATA_FILE})
+
+        COMPONENTS_LENGTH=$(jq '.components |length' ${SNAPSHOT_PATH})
+
+        COSIGN_SIG_KEY=$(fetch_cosign_key $CONFIG_MAP_NAME)
+
+        for (( COMPONENTS_INDEX=0; COMPONENTS_INDEX<COMPONENTS_LENGTH; COMPONENTS_INDEX++ )); do
+            INTERNAL_CONTAINER_IMAGE=$(jq -r ".components[${COMPONENTS_INDEX}].containerImage" ${SNAPSHOT_PATH})
+            INTERNAL_CONTAINER_REF="${INTERNAL_CONTAINER_IMAGE/@*}"
+            DIGEST="${INTERNAL_CONTAINER_IMAGE/*@}"
+
+            PUB_CONTAINER_REF=$(jq -r ".components[${COMPONENTS_INDEX}].repository" ${SNAPSHOT_PATH})
+            PUB_CONTAINER_IMAGE="${PUB_CONTAINER_REF}@${DIGEST}"
+
+            COSIGN_PRE_ARGS="AWS_REGION=${AWS_REGION}"
+            COSIGN_PRE_ARGS="$COSIGN_PRE_ARGS AWS_ACCESS_KEY_ID=${AWS_ACCESS_KEY_ID}"
+            COSIGN_PRE_ARGS="$COSIGN_PRE_ARGS AWS_SECRET_ACCESS_KEY=${AWS_SECRET_ACCESS_KEY}"
+            COSIGN_COMMON_ARGS="--key ${COSIGN_SIG_KEY}"
+
+            IMAGE=$(skopeo inspect --raw "docker://${INTERNAL_CONTAINER_IMAGE}")
+            MEDIA_TYPE=$(echo "$IMAGE" | jq -r '.mediaType')
+            LIST=0
+            if [ "$MEDIA_TYPE" = "application/vnd.docker.distribution.manifest.list.v2+json" ]; then LIST=1; fi
+            if [ "$MEDIA_TYPE" = "application/vnd.oci.image.index.v1+json" ]; then LIST=1; fi
+
+            # if manifest list, sign each manifest + ml itself
+            if [ $LIST -eq 1 ]; then
+              PUB_REPO_REF="${PUB_CONTAINER_REF}@${DIGEST}"
+              for MDIGEST in $(echo "$IMAGE" | jq -rc '.manifests[]|.digest'); do
+                cosign ${COSIGN_PRE_ARGS} cosign -t 3m0s sign\
```

Comment:
This looks like a typo. I don't think `cosign cosign sign` is the correct command - should it be just `cosign sign`? https://github.com/sigstore/cosign/blob/main/doc/cosign_sign.md

And also, I'm curious about the `COSIGN_PRE_ARGS` variable. It will contain `AWS_REGION=...` etc. But these are not valid `cosign` argument, are they? Or are they meant to be env vars supplied to `cosign` (that would mean that the leading `cosign` shouldn't be there)? In that case it would be redundant, because they supplied to everything that is invoked in this script

Reply:
yeah, there's typo, COSIGN_PRE_ARGS are env variables needed for cosign (if you're using AWSKMS). So in practice envs are not needed to be set explicitly, but it just feels better to explicitly put in here to emphasize they are needed. But I can remove them if you insist

Reply:
fixed

Reply:
Thanks.